woocommerce/woocommerce-ios
# Google Ads: Update UI for Google Ads dashboard card

Part of #13230 


## Description

This PR updates the layout of the Google Ads dashboard card. Changes include:
- Updated the card view model:
  - Fetch last campaign's details. Since the REST API doesn't support pagination, we're fetching all campaigns and get the last active (enabled) one. If no active campaigns are found, we display the last disabled one.
  - Retrieve last campaign's stats in the background. I'm currently fetching all stats but asking the plugin team if we can fetch stats for a specific campaign only (p1720153987272339-slack-C02BB3F30TG).
- Added layout for campaign details and stats. I'm using the same layout with Blaze's for simplicity. Stats will not be displayed if they cannot be loaded.
- Handled error loading states.
- Handled tap on the dismiss button.
- Updated wording for the empty state.
- Added generic tracking events related to dynamic dashboard.

Due to the large size of the PR, unit tests and navigation for the card will be handled in a separate PR.

## Testing steps

- Pre-requisite: Install, activate, and set up the Google Ads plugin to a test site following pe5sF9-2Qo-p2. Ensure that you're using the plugin version 2.7.5 or above.
- Build the app and log in to your store. 
- Enable Google Ads card if you haven't already. Confirm that the card shows the loading state when fetching data. Confirm that `dynamic_dashboard_card_data_loading_started` with `type: google-ads` is tracked.
- When data loading completed, confirm that `dynamic_dashboard_card_data_loading_completed` with `type: google-ads` is tracked
- If your store doesn't have any campaign, confirm that the empty state is displayed.
- Create a Google ads campaign for your store in wp-admin or from the Hub menu.
- Pull to refresh the dashboard screen and confirm that the campaign details are displayed.
- Disconnect the internet connection and reload the dashboard.
- Confirm that the error state is displayed and `dynamic_dashboard_card_data_loading_failed` with `type: google-ads` and error details is tracked.
- Tap on Retry, confirm that `dynamic_dashboard_card_retry_tapped` is tracked with `type: google-ads`.
- Tap on the ellipsis menu > Hide Google Ads. Confirm that Customize screen is displayed and `dynamic_dashboard_hide_card_tapped` is tracked with `type: google-ads`.

## Screenshots

Loading state | Empty state | Campaign details | Error state
--- | --- | --- | ---
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/e157b0f1-1f9c-45ea-ad1d-8a2a8ed57538" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/76e4ac0b-69d3-4630-bc71-d557556bc4e2" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/1354c275-57dc-48d1-ae84-f54bc8a30570" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/3cb6de27-04e0-4b5c-9fe9-f8365a6e7491" width=320 />

---
